### DISCUSSION TITLE
Does jRelease provide a way to increment versions?

### DISCUSSION THREAD

**Discussion by nazeem-soeltan-maersk** [jreleaser/jreleaser#880] [Q&A]
Hi,

I'm trying out jReleaser and I have a simple PoC ready which publishes a single jar into the repository and sends a slack notification. We are using Maven and within gitlab CI we are calling the maven jreleaser command.
I'm wondering, and I'm not entirely sure, but does jReleaser provide a way to increment versions (minor, major version bump) somewhat out of the box, or are other tools required. If so, are there any 'proven' examples (maybe by other community members) where maven (plugins) are used or any other way?

**Comment by aalmiray** [comment]
Not at the moment, it does not, and for various reasons.

JReleaser can be executed in 4 ways: CLI, Maven plugin, Gradle plugin, Ant tasks. One might think that the Maven and Gradle plugins could provide this feature however, that would leave CLI and Ant task hanging out to dry. Why? Because there's no standard convention to define where the project version should be stored for CLI and Ant projects, it's even a stretch for Gradle as there are many places where that info could come from before it's attached to the project's `version` property.

Next, JReleaser's commands are supposed to be separate from build commands, that is, you use them only to release a project (with or without binaries). The build process is the one selecting the version to be released. Maven and Gradle already have plugins that provide version bumping capabilities (Maven has the [release](https://maven.apache.org/maven-release/maven-release-plugin/) and [versions](https://www.mojohaus.org/versions-maven-plugin/) plugins). In theory, JReleaser could calculate the next version, specially if [SemVer](https://semver.org/) and [Conventional-Commits](https://www.conventionalcommits.org/en/v1.0.0/) are followed to the letter, but not every project does and not every project should or will. Even if JReleaser could calculate the next version the problem would be storing it back depending on the usage mode (CLI, Maven, Gradle, Ant).

At best what JReleaser _could_ do in the future is suggest the next version number given certain rules and constraints, write it out as part of the `output.properties` file, then let the build tool of choice read the file and update the project's metadata accordingly.
